docs/.vitepress/components/SnippetToggler.vue

```diff
@@ -0,0 +1,117 @@
+<script lang="ts" setup>
+import { ref } from 'vue';
+
+const props = defineProps<{
+	choices: string[];
+	initial: string;
+	label?: string;
+}>();
+
+const pref = ref(props.initial);
+
+const onPrefChange = (event: Event) => {
+	pref.value = (event.target as HTMLSelectElement).value;
+};
+</script>
+
+<template>
+	<div class="snippet-toggler">
+		<div class="snippet-toggler-header">
+			<span class="snippet-toggler-header-label">{{ label }}</span>
+
+			<span class="spacer" />
+
+			<div class="snippet-toggler-header-lang-container">
+				<select class="snippet-toggler-header-lang" :value="pref" @change="onPrefChange">
```

Comment:
Vue has a convenient shorthand for this type of two-way binding 🙂 

```suggestion
				<select class="snippet-toggler-header-lang" v-model="pref">
```

Reply:
I tried this and it didn't work for some reason so used this method